src/storage/data_table.cpp

```diff
@@ -59,6 +61,54 @@ void DataTable::Scan(const common::ManagedPointer<transaction::TransactionContex
   out_buffer->SetNumTuples(filled);
 }
 
+void DataTable::NUMAScan(common::ManagedPointer<transaction::TransactionContext> txn,
+                         std::vector<ProjectedColumns *> *out_buffers, ProjectedColumns *const result_buffer,
+                         common::ExecutionThreadPool *thread_pool) {
+  std::vector<numa_region_t> numa_regions;
+  GetNUMARegions(&numa_regions);
+
+  // TODO(emmanuel) add additional buffers to the array so that the assert below is unnecessary
+
+  TERRIER_ASSERT(out_buffers->size() >= numa_regions.size(),
+                 "should have at least as many outbuffers as numa regions currently occupied");
+  for (uint32_t i = 0; i < numa_regions.size(); i++) {
+    // Lambda function to pass into thread pool
+    ProjectedColumns *out_buffer = (*out_buffers)[i];
+    numa_region_t region = numa_regions[i];
+    auto lambda = [&, out_buffer, region] {
+      uint32_t filled = 0;
+
+      for (auto it = begin(region); it != end(region) && filled < out_buffer->MaxTuples(); it++) {
```

Comment:
The scan will stop if it reaches the end of the region or if the out_buffer is filled. If the latter case happens, how can the caller know where to pick up?  The original scan doesn't have this problem because it used SlotIterator start_pos as input to enable the scan starting on a specific slot if the buffer is filled in the previous round.

Reply:
Yeah, this is an issue, we'll discuss solutions in the next few weeks.

Reply:
We no longer need this function so it has been deleted